src/cucumber-json.js

```diff
@@ -1,62 +1,61 @@
-var shell = require('shelljs'),
-config;
+var shell   = require('shelljs'),
+    compose = require('./compose');
 
-function cucumberJSON(d) {
-    return new Promise(function(resolve, reject){
-        var jsonCucumber;
-        var shellOut = shell.exec(
-            command(d),
-            { silent: true, async: false }
-        );
-        if (shellOut.stderr) {
-            reject(shellOut.stderr);
-        }
-        jsonCucumber = JSON.parse(shellOut.stdout);
-        resolve(
-            {
-                [d]: {
-                    testType: config.testType || 'all tests',
-                    count: totalNumberofTests(jsonCucumber),
-                    jsonCucumber: jsonCucumber
-                }
+function fetch(config) {
+    return Promise
+        .all(promises(config))
+        .then(function(arrayOfCucumberJSON) {
+            return arrayOfCucumberJSON;
+        })
+        .catch(function(error) {
+            return error;
+        });
+}
+
+function promises(config) {
```

Comment:
@cookiescrumbs I inlined the promise and the looping to return an array of promises, not sure if this is any better tbh